src/readwrite.rs

```diff
@@ -0,0 +1,91 @@
+// Copyright 2018-2019 Mozilla
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+use lmdb::{
+    Database,
+    RoCursor,
+    RoTransaction,
+    RwTransaction,
+    Transaction,
+    WriteFlags,
+};
+
+use crate::error::StoreError;
+use crate::read_transform;
+use crate::value::Value;
+
+pub struct Reader<'env>(pub RoTransaction<'env>);
+pub struct Writer<'env>(pub RwTransaction<'env>);
+
+pub trait Read {
```

Comment:
As you mentioned about the naming, I think `Reader&Writer` are reasonable names for read_only/write transactions. Though I find the `Read` trait a bit confusing here. At the first glance, I thought this is a reader specific trait. I believe that we wanted to express the fact that both read_only and write transactions *can* read the store.

Would it be better to name this trait as `Readable`, despite that it's not the preferred naming convention for traits in Rust?

Reply:
Yes, it's a good point, the trait name is too confusing. I've renamed it to *Readable*.

Reply:
For what it's worth, it isn't clear to me what the preferred naming convention is for traits in Rust. https://github.com/milesmcc/ieql/issues/2 suggests that it's "imperatives over modified words," in which case *Read* would be preferred to *Readable*.

But https://github.com/rust-lang-nursery/api-guidelines/issues/28 suggests that the matter is more complicated, and it isn't clear to me that the answer is as simple as to use imperatives. In any case, it's plenty reasonable for us to use *Readable*, especially for a trait that consumers usually won't interact with directly.